internal/core/src/segcore/SegmentGrowingImpl.cpp

```diff
@@ -603,6 +618,33 @@ SegmentGrowingImpl::bulk_subscript(FieldId field_id,
     return result;
 }
 
+void
+SegmentGrowingImpl::bulk_subscript_sparse_float_vector_impl(
+    FieldId field_id,
+    const ConcurrentVector<SparseFloatVector>* vec_raw,
+    const int64_t* seg_offsets,
+    int64_t count,
+    milvus::proto::schema::SparseFloatArray* output) const {
+    AssertInfo(HasRawData(field_id.get()), "Growing segment loss raw data");
+
+    // if index has finished building index, grab from index
+    if (indexing_record_.SyncDataWithIndex(field_id)) {
```

Comment:
could there be a race here?

1. we check syncdata and it is a not synced
2. insert thread build index and remove the chunk
3. when we try to retrieve from chunk, the original data has already been removed

Reply:
I thought bulk_subscript_impl has same problem

Reply:
this is related to both dense and sparse, I am updating it in https://github.com/milvus-io/milvus/pull/31170.